### ISSUE TITLE
Incorrect suggestion for return type :void

### ISSUE THREAD

**Issue by komandakycto** [kalessil/phpinspectionsea#1489]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Plugin         | Php Inspections (EA Extended), 4.0.3 |
| Language level | PHP 7.1                        |

Current behaviour
--
I have a simple method, plugin suggest me set return type as void but method has return statement.
```php
    /**
     * @param $response
     * @return null
     */
    protected function handleResponse($response)
    {
        if (empty($response) || $this->transport->getStatusCode() !== 200) {
            return null;
        }

        return $response;
    }
```
![Screenshot from 2020-01-15 14-02-57](https://user-images.githubusercontent.com/4821109/72429696-9f4b5c80-37a1-11ea-92ff-e7cc1be47a94.png)

Expected behaviour
--
Don't suggest anything because we can't determine return variable type in this situation 

Environment details
---
PhpStorm 2019.3.1
Build #PS-193.5662.63, built on December 19, 2019
Runtime version: 11.0.5+10-b520.17 amd64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
Linux 4.15.0-74-generic
GC: ParNew, ConcurrentMarkSweep
Memory: 951M
Cores: 4
Registry: run.processes.with.pty=TRUE
Non-Bundled Plugins: com.kalessil.phpStorm.phpInspectionsEA, lv.midiana.misc.phpstorm-plugins.deep-keys

**Comment by orklah** [comment]
Hi

`@param $response` PHPStorm don't know the type of this variable. When trying to check the return type, this will be ignored, so the only type is null.

If you document the $response type(even with mixed), It will calculate the return type correctly and stop suggesting void

**Comment by komandakycto** [comment]
Thanks for answer! I think it looks strange, we know that method always return null or something but plugin suggest void. 
When I set types in phpDoc section plugin still suggest me void as return type. Only If I modify `@return` tag in phpDoc section to `@return array|null` plugin suggestion will be correct.
![Screenshot from 2020-01-15 16-38-42](https://user-images.githubusercontent.com/4821109/72438661-5a7df080-37b6-11ea-9951-99f1cd2bddf4.png)

 It seems to phpDoc has more highest priority than source code.
![Screenshot from 2020-01-15 16-45-18](https://user-images.githubusercontent.com/4821109/72438694-6f5a8400-37b6-11ea-9c24-8141be9e2926.png)

**Comment by kalessil** [comment]
We are relying to PhpStorm types resolver here, so marking as 3rd party.
